.travis.yml

```diff
@@ -2,19 +2,17 @@ language: go
 
 go_import_path: github.com/pingcap/tidb
 
-go:
-  - "1.12"
 
-env:
-    - TRAVIS_COVERAGE=0
-    - TRAVIS_COVERAGE=1
-
-# Run coverage tests.
 matrix:
+  include:
+    - go: "1.12"
+      env: TRAVIS_COVERAGE=0 TEST_SUITE="check"
+    - go: "1.12"
+      env: TRAVIS_COVERAGE=0 TEST_SUITE="check-fail"
   fast_finish: true
   allow_failures:
     - go: "1.12"
-      env: TRAVIS_COVERAGE=1
+      env: TRAVIS_COVERAGE=0 TEST_SUITE="check-fail"
```

Comment:
why allowing the "check-fail" failure?

Reply:
Because now, the check-fail cannot be passed.
https://travis-ci.org/pingcap/tidb/jobs/506081661

Reply:
how about not running this check? It can help us save CI resources and CI time?

Reply:
Because of [fast finish](https://docs.travis-ci.com/user/build-matrix/#fast-finishing), I think open this is not a problem in CI time.
And open this can warn us to pay attention to some check.
Needless to say, after this PR, CI will only run `make check`, it is not the bottleneck in time.

Reply:
Can we remove CI? Seems "check_dev" already did the test and check work.

Reply:
Of course, we could.
But I think Travis-ci has a good advantage which is `allow_failure`.
We could use `aloow_failure` to test the newest golang version, and smooth upgrade golang.